clang/lib/Sema/Sema.cpp

```diff
@@ -687,6 +687,48 @@ void Sema::diagnoseZeroToNullptrConversion(CastKind Kind, const Expr *E) {
       << FixItHint::CreateReplacement(E->getSourceRange(), "nullptr");
 }
 
+void Sema::DiagnoseAssignmentBoolContext(Expr *E, QualType Ty) {
+  // Use copy to not alter original expression.
```

Comment:
This is unnecessary.  You don't use the original value of `E` anywhere else below, so there isn't a good reason to  not just use the parameter.